docs/proposals/virtualhost-policy.md

```diff
@@ -0,0 +1,98 @@
+# VirtualHost policy
+
+## Context
+
+Since Kuma 1.1.4 the dataplane now embeds a CoreDNS instance which forwards all DNS traffic to envoy proxy.
+This opens the door to more flexible implementation of vips.
+We propose here to create a new policy which will enable users to create arbitrary hostnames and ports that will route to a specific sets of tags.
+
+## New Policy
+
+We will add a new policy type:
+
+```yaml
+type: VirtualHost
+mesh: string # The mesh this policy applies to
+name: string #The name of the policy
+tags:
+    # A map of tags that should be taken for creating the tag selector. 
+    # It can either have a value or have "*" in which case a different vip will be generated for each tuple
+host: string # A go template with a special function `tag` which can use tags present in the `tags` section.
+port: int # A port to expose the virtual host on
+```
+
+It is impossible to create a VirtualHost that hasn't got a tag `kuma,io/service`.
+
+### Example
+
+Simple service with no wildcard replacement:
+
+```yaml
+type: VirtualHost
```

Comment:
I'm a little bit hesitant here on the name. Envoy uses `VirtualHost` in the configuration for HTTP traffic. This might be confusing for someone more advanced and familiar with Envoy.

a couple of ideas: `VirtualDestination`, `VirtualDNS`? I cannot find a good name, maybe the rest of the team will have a better idea.

Reply:
Haha seems like me when I wrote the proposal :)
VirtualDNS doesn't sound right as we're adding ports as well.
VirtualOutbound doesn't sound right to me either as it's not really an outbound.
VirtualDestination is very vague but seems the least wrong of all our ideas.

Let's see if someone brings something better.

Reply:
VirtualEndpoint?

Reply:
VirtualHostPort?